lib/ProductOpener/Import.pm

```diff
@@ -412,6 +412,167 @@ my %no = (
+			$stats{orgs_without_source_authorization}{$org_id}++;
+			next;
+		};
+	}
+
+	# The do_not_import_codeonline checkbox will be replaced by the new system above that will work for all sources
+
+	# Check if it is a CodeOnline import for an org with do_not_import_codeonline
+	if ((defined $args_ref->{source_id}) and ($args_ref->{source_id} eq "codeonline")
+		and (defined $org_ref->{do_not_import_codeonline}) and ($org_ref->{do_not_import_codeonline})) {
+		$log->debug("skipping codeonline import for org with do_not_import_codeonline", { org_ref => $org_ref }) if $log->is_debug();
+		next;
+	}
+	return;
+}
+
+
+
+sub get_org_id_for_product($$$$$$) {
+	
+	my $imported_product_ref = shift;
+	my $org_id = shift;
+	my $org_ref = shift;
+	my %stats = shift;
+	my $args_ref = shift;
+	my $glns_ref = shift;
+
+	$org_id = get_string_id_for_lang("no_language", $imported_product_ref->{org_name});
+	if ($org_id eq "") {
+		return;
+	}
+
+	# Re-assign some organizations
+	# e.g. nestle-france-div-choc-cul-bi-inf -> nestle-france
+	$org_id =~ s/^nestle-france-.*/nestle-france/;
+	$org_id =~ s/^cereal-partners-france$/nestle-france/;
+	$org_id =~ s/^nestle-spac$/nestle-france/;
+
+	# organizations by GLN
+
+	my %gln = (
+		"3010337111109" => "nestle-france",
+		"3012789200103" => "nestle-france",	# SPAC (Buitoni)
+		"3011542300012" => "nestle-france",	# Herta
+		"3013873929306" => "nestle-france",	# Cereal Partners
+		"3011797320001" => "nestle-france",	# Nestlé Waters
+		"3010737870309" => "groupe-bel",	# BEL FRANCE
+	);
+
+	# GLNs can now be stored inside organization profiles (loaded in $glns_ref)
+
+	if (defined $imported_product_ref->{"sources_fields:org-gs1:gln"}) {
+		if ($glns_ref->{$imported_product_ref->{"sources_fields:org-gs1:gln"}}) {
+			$org_id = $glns_ref->{$imported_product_ref->{"sources_fields:org-gs1:gln"}};
+		}
+		elsif (defined $gln{$imported_product_ref->{"sources_fields:org-gs1:gln"}}) {
+			$org_id = $gln{$imported_product_ref->{"sources_fields:org-gs1:gln"}};
+		}
+	}
+
+	$log->debug("org", { org_name => $imported_product_ref->{org_name}, org_id => $org_id, gln => $imported_product_ref->{"sources_fields:org-gs1:gln"} }) if $log->is_debug();
+
+	defined $stats{orgs_in_file}{$org_id} or $stats{orgs_in_file}{$org_id} = 0;
+	$stats{orgs_in_file}{$org_id}++;
+
+	$org_ref = retrieve_org($org_id);
+
+	if (defined $org_ref) {
+
+		defined $stats{orgs_existing}{$org_id} or $stats{orgs_existing}{$org_id} = 0;
+		$stats{orgs_existing}{$org_id}++;
+
+		# Make sure the source as the authorization to load data to the org
+		# e.g. if an org has loaded fresh data manually or through Equadis,
+		# don't overwrite it with potentially stale CodeOnline or USDA data
+
+		apply_organization_settings_to_product(%stats, $org_id, $org_ref, $args_ref);
+
+		# If it is a GS1 import (Equadis, CodeOnline, Agena3000), check if the org is associated with known issues
```

Comment:
The code below should go in a function named apply_organization_settings_to_product() (different from the one that you added where it checks the authorization).